sysfs/class_drm.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package sysfs
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"syscall"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+var (
+	// list of supported device drivers.
+	drmDeviceDrivers = []string{"amdgpu"}
+)
+
+// ClassDrmCardStats contains info from files in /sys/class/drm/card<card>/device
+// for a single card.
+// https://www.kernel.org/doc/html/latest/gpu/drivers.html
+// https://www.kernel.org/doc/html/latest/gpu/amdgpu.html
+type ClassDrmCardStats struct {
```

Comment:
nit: Acronyms like DRM should be uppercase: https://github.com/golang/go/wiki/CodeReviewComments#initialisms